ENA-3339

haseeb-gh opened the pull request:
- Added support for generating BAM files from fastqs.
- Added a utility that fixes fastqs files that have Uracil bases in them.

haseeb-gh committed:
ENA-3339

- Added support for generating BAM files from fastqs.
- Added a utility that fixes fastqs files that have Uracil bases in them.

haseeb-gh commented on resources/fastq2sam/fastq2sam.sam:
```diff
@@ -0,0 +1,4 @@
+@HD	VN:1.6	SO:queryname
+@RG	ID:A	SM:sample001
+rn001	77	*	0	0	*	*	0	0	GACGGATCTATAGCAAAACT	''))))((*+**+*(**)*+	RG:Z:A
+rn001	141	*	0	0	*	*	0	0	GACGGATCTATAGCAAAACT	''))))((*+**+*(**)*+	RG:Z:A
```
this SAM file generated by picard converted the quality scores from the source fastq files:
FFHHHHGGIJIIJIGIIHIJ

into:
''))))((*+**+*(**)*+

@raskoleinonen any idea why this might have happened?

raskoleinonen commented on src/main/java/uk/ac/ebi/ena/readtools/fastq/Fastq2Sam.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    public void convert() {
+        new picard.sam.FastqToSam().instanceMain(new String[]{
+                "SM=" + sampleName,
+                "F1=" + inputFile1,
+                "F2=" + inputFile2,
+                "O=" + outputFile
+        });
```
I wish all coding was as simple as this.

raskoleinonen commented on src/main/java/uk/ac/ebi/ena/readtools/utils/Utils.java:
```diff
@@ -0,0 +1,53 @@
+
+import htsjdk.samtools.fastq.AsyncFastqWriter;
+import htsjdk.samtools.fastq.BasicFastqWriter;
+import htsjdk.samtools.fastq.FastqReader;
+import htsjdk.samtools.fastq.FastqRecord;
+import htsjdk.samtools.fastq.FastqWriter;
```
Nice that samtools also has FastqReader/Writer we can use.

haseeb-gh committed:
ENA-3339

- minor changes to api docs.

haseeb-gh committed:
ENA-3339

- added the ability to determine replacement T base case based on the source U base.
